### DISCUSSION TITLE
Onscreen overlay "On-active only" option

### DISCUSSION THREAD

**Discussion by gitmaker032** [SaifAqqad/AHK_MicMute#35] [Ideas]
Hello! Thanks for the great app.

But I really miss the "On-active only" (Onscreen overlay) option. For me it is more important to know when the microphone is active. A kind of explicit reminder that the microphone is active and should not forget to turn it off.

I think it would also be nice to add a semi-transparent black background for the icon, because it can blend in with the white objects on the screen now.

![image](https://user-images.githubusercontent.com/67278783/132729882-5e245fd9-e46a-4263-a20e-c97fc81dab95.png)

**Comment by SaifAqqad** [comment]
> But I really miss the "On-active only" (Onscreen overlay) option. For me it is more important to know when the microphone is active. A kind of explicit reminder that the microphone is active and should not forget to turn it off.

I've added this in commit e953182a9de9f3b745cb233368626a8856ca78a5, you can try it out in the development build [here](https://github.com/SaifAqqad/AHK_MicMute/releases/tag/latest_commit).


> I think it would also be nice to add a semi-transparent black background for the icon, because it can blend in with the white objects on the screen now.

I'm sorry but I won't implement this for 2 reasons
1: The icon already has a gray outline that makes it visible on white backgrounds -> ![image](https://user-images.githubusercontent.com/47293197/135900029-05cef5fc-5550-47f3-a1e2-07cce0cefc83.png)

2: Setting the transparency level for the overlay window would also affect the icon, so I would need to rewrite the overlay as two separate windows, one for the icon and the other for the background.

**Reply by gitmaker032** [reply]
> I've added this in commit e953182, you can try it out in the development build here.

It works great, thank you! It's also good that the bug with the mute at startup was fixed.

>The icon already has a gray outline that makes it visible on white backgrounds

It still blends heavily with the white elements (of which there are many in the dark theme of various software). That is easy to lose sight of the icon. Dark background would have fixed the situation, I suppose.

![image](https://user-images.githubusercontent.com/67278783/137868183-120991c5-eedb-46a3-b0e3-0bf2218d9a5e.png)

>2: Setting the transparency level for the overlay window would also affect the icon, so I would need to rewrite the overlay as two separate >windows, one for the icon and the other for the background.

How about using a single png image with the necessary transparency?

**Reply by SaifAqqad** [reply]
Since the overlay already supports setting a custom icon, you can just turn that option on and put these two icons in the same folder as `MicMute.exe`, they're the same as the default icons but with a black background (70% opacity).

`overlay_mute.png`
![overlay_mute](https://user-images.githubusercontent.com/47293197/137940907-6ebef75b-f482-4068-bd33-7ad444348431.png)
`overlay_unmute.png`
![overlay_unmute](https://user-images.githubusercontent.com/47293197/137940942-cb43d1c4-cc4e-400d-9a86-fdbd5f870319.png)

**Reply by gitmaker032** [reply]
Thank you! It's much better now. But the icon is displayed too small. Maybe it would be a good idea to add the relevant parameter to the config.json. Something like this:
```
"OverlayCustomIconsSizes": {
"w": 256,
"h": 256
}
```

![image](https://user-images.githubusercontent.com/67278783/138344758-a8cb43b6-971d-40ba-9ec3-9917a3298afa.png)

**Comment by cloaked2222** [comment]
@SaifAqqad  When you hover over the "Use custom icons" button, please add a help-tip with instructions on where to setup the icons and what size/type of icon to use. It was a mystery to me as a user, until I found this discussion.

**Reply by SaifAqqad** [reply]
Sure I'll add a tooltip for it in the next version, though It's actually mentioned in the [README](https://github.com/SaifAqqad/AHK_MicMute#3-on-screen-overlay).

**Reply by cloaked2222** [reply]
Ditto for "Use custom sounds". Even if the popup just says 'see readme', haha. The more intuitive you can make it for users the better. Anyways, thanks for making this!
